src/main/java/com/globallogic/amcr/persistence/mapper/librarycomponent/BookMapper.java

```diff
@@ -0,0 +1,81 @@
+package com.globallogic.amcr.persistence.mapper.librarycomponent;
+
+import com.globallogic.amcr.persistence.model.librarycomponent.Book;
+import com.globallogic.amcr.typehandler.UUIDTypeHandler;
+import org.apache.ibatis.annotations.*;
+import org.apache.ibatis.type.Alias;
+
+import java.util.List;
+import java.util.UUID;
+
+@Mapper
+@Alias(value = "UUIDTypeHandler")
+public interface BookMapper {
+
+    /**
+     * Method for saving new book into the database.
+     * @param book - Book Object containing data for the new book entry into the database.
+     */
+    @Insert("INSERT INTO library(id, title, genre, author, cover, email, reader, available) VALUES (#{id, javaType=java.util.UUID, jdbcType=OTHER, typeHandler=UUIDTypeHandler}, #{title}, #{genre}, #{author}, #{cover}, #{email}, #{reader}, #{available})")
+    void save(Book book);
+
+    /**
+     * Method that allows you to retrieve one individual book.
+     * @param id - UUID book id for identifying book.
+     * @return - Book Object containing all the information for selected book.
+     */
+    @Results(id = "bookSingleResult")
+    @ConstructorArgs({
+            @Arg(column = "id", javaType = UUID.class, typeHandler = UUIDTypeHandler.class, id = true),
+            @Arg(column = "title", javaType = String.class),
+            @Arg(column = "genre", javaType = String.class),
+            @Arg(column = "author", javaType = String.class),
+            @Arg(column = "reader", javaType = String.class),
+            @Arg(column = "available", javaType = Boolean.class),
+            @Arg(column = "cover", javaType = String.class),
+            @Arg(column = "email", javaType = String.class)
+    })
+    @Select("SELECT * FROM library WHERE #{id, javaType=java.util.UUID, jdbcType=OTHER, typeHandler=UUIDTypeHandler} = id")
+    Book get(@Param("id") UUID id);
+
+    /**
+     * Method that allows you to retrieve a list of books and content from the database.
+     * @return - List of books.
+     */
+    @Results(id = "bookAllResults")
```

Comment:
This can just be changed to `@ResultMap("bookSingleResult")` as the map has already defined and the id was set in the annotations of the previous method